aten/src/ATen/native/cuda/CUDALoops.cuh

```diff
@@ -237,16 +272,22 @@ at::opmath_type<f_inputs_type> scalar_val) {
     }
   }
 
+  auto extra_args_array = parameter_pack_to_arg_array(extra_args...);
+  // size of `extra_args` is known at compile-time
+  constexpr auto extra_args_size = sizeof...(extra_args);
+
   if (vectorized) {
-    std::array<void*, 7> args = {
-      (void*)&N,
-      (void*)&data,
-      (void*)&scalar_val,
-      nullptr,
-      nullptr,
-      nullptr,
-      nullptr
-    };
+    // pack args for kernel launch
+    constexpr int kernel_args = 3;
+    void* args[kernel_args + extra_args_size];
+    args[0] = static_cast<void*>(&N);
+    args[1] = static_cast<void*>(&data);
+    args[2] = static_cast<void*>(&scalar_val);
+
+    for (size_t i = 0; i < extra_args_size; i++) {
```

Comment:
can you try using c10::irange here, there are internal errors for this line:
```
/caffe2/aten/src/ATen/native/cuda/CUDALoops.cuh(287): warning: pointless comparison of unsigned integer with zero
```

Reply:
@ngimel and I reviewed more closely and it looks like internal warnings, not errors, but we should still try irange here

Reply:
Done :)